File: SIN BDD/RESTFULL_DOTNET_CONUNI_GR06/06. CLIWEB/cliente_web_rest_dotnet_conuni_gr06/src/Servicio/ServicioTemperatura.cs
using Ec.Edu.Monster.Modelo;
using Ec.Edu.Monster.Utils;

namespace Ec.Edu.Monster.Servicio;

public class ServicioTemperatura
{
    private readonly ClienteRest cliente = new();

    public Resultado CelsiusAFahrenheit(double celsius) => cliente.Convertir("temperatura", "celsiusAFahrenheit", celsius);
    public Resultado FahrenheitACelsius(double fahrenheit) => cliente.Convertir("temperatura", "fahrenheitACelsius", fahrenheit);
    public Resultado CelsiusAKelvin(double celsius) => cliente.Convertir("temperatura", "celsiusAKelvin", celsius);
    public Resultado KelvinACelsius(double kelvin) => cliente.Convertir("temperatura", "kelvinACelsius", kelvin);
    public Resultado FahrenheitAKelvin(double fahrenheit) => cliente.Convertir("temperatura", "fahrenheitAKelvin", fahrenheit);
}

File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/08. SERVIDOR/servidor_restful_java_conuni_gr06/src/java/ec/edu/monster/servicio/ServicioTemperatura.java
package ec.edu.monster.servicio;

public class ServicioTemperatura {

    public double celsiusAFahrenheit(double celsius) {
        return (celsius * 9.0 / 5.0) + 32.0;
    }

    public double fahrenheitACelsius(double fahrenheit) {
        return (fahrenheit - 32.0) * 5.0 / 9.0;
    }

    public double celsiusAKelvin(double celsius) {
        return celsius + 273.15;
    }

    public double kelvinACelsius(double kelvin) {
        return kelvin - 273.15;
    }

    public double fahrenheitAKelvin(double fahrenheit) {
        return (fahrenheit - 32.0) * 5.0 / 9.0 + 273.15;
    }
}
